### ISSUE TITLE
TBG: Windows (&Mac) EOL Rejected

### ISSUE THREAD

**Issue by ax3l** [ComputationalRadiationPhysics/picongpu#1775]
I got a user bug report yesterday, that files edited under windows - which naturally get converted to `CR` `LF` line endings, are catched by our tbg tool via the ["contains spaces after line continuation"](https://github.com/ComputationalRadiationPhysics/picongpu/blob/0.2.2/src/tools/bin/tbg#L379) check.

We should evaluate if that is really critical, if we output a more sophisticated error message or if we do a conversion.

Also, we should write out `sed` commands as in our style checks that can be used by the user to copy-&-paste-fix the file in concern.

### Note on EOL
- Windows: `CR` `LF`
- Unix: `LF`
- Mac: `CR`

**Comment by theZiz** [comment]
Maybe we should use `[:blank:]` instead of `[:space:]`: http://www.regular-expressions.info/posixbrackets.html
`[:space:]` is: `[ \t\r\n\v\f]`, whereas `[:blank:]` is only `[ \t]`. In other words: Our check searches for a line break in the line (`CR` for Windows/Mac) and afterwards finds something, which matches for `[:space:]` (the `\r`). I have no idea, what `\v` and `\f` do and whether they're important. ;)

**Comment by ax3l** [comment]
yep, feel free to open a PR!

> Also, we should write out `sed` commands as in our style checks that can be used by the user to copy-&-paste-fix the file in concern.

Please [add at the same PR](https://github.com/ComputationalRadiationPhysics/picongpu/blob/ef696376e52338161a6484d09b3b8f616d184998/test/hasEOLwhiteSpace#L58-L64), in a 2nd commit to the output:


```bash
  echo "Run the following command(s) on the above files to remove your"
  echo "end-of-line (EOL) white spaces:"
  echo ""
  for i in ${files[@]}
  do
    echo "sed -i 's/[[:blank:]]\+$//' $i"
  done
```

**Comment by theZiz** [comment]
Okay, `\v` is a vertical tab (lol?) and `\f` is [Form Feed](https://en.wikipedia.org/wiki/Form_Feed). I guess we can just ignore them. Or if **really** needed check for `[ \t\v\f]` instead of `[:space:]`...

**Comment by ax3l** [comment]
Well, the actual question is: *what characters are allowed after bash line continuation*?

Because I found a random comment on [SO where a user actually got no bash line-continuation with his Windows file](http://stackoverflow.com/questions/3871332/how-to-tell-bash-that-the-line-continues-on-the-next-line#comment28877980_3871336). Needs validation.

**Comment by theZiz** [comment]
I guess every character is allowed which compiles which is handled by the CI check?

**Comment by theZiz** [comment]
Created PR: https://github.com/ComputationalRadiationPhysics/picongpu/commit/020b125dd85046fde0fe19481a3f4c67ccd5e813

**Comment by psychocoderHPC** [comment]
closed with #1777
